### DISCUSSION TITLE
How do you create a custom AVD (Android Virtual Device) other than the forced default of a Pixel 7?

### DISCUSSION THREAD

**Discussion by ghost** [beeware/briefcase#2010] [Q&A] [answered]
I'm making an app that isn't intended for mass consumption and will only be run on a specific device that I built from a raspi, running Android 14.

I just want to make a virtual device with Briefcase that has a custom display res/dpi and based on Android 14. However, when I make a device with briefcase, it forces me to use Pixel 7 and I can't change anything about it's main display. I can only add secondary displays. I also can't seem to change anything about the Android image.

Does this have to be done by installing the full on Android SDK IDE, or can I do it in briefcase somehow? Further, how do I manage the devices that get created with briefcase (e.g., delete a device)?

EDIT: I found this page in the docs after not seeing anything related in the help commands.
https://briefcase.readthedocs.io/en/latest/reference/platforms/android/gradle.html#additional-options

Looks like I can at least specify things in some way, but now I'm confused on how to fill the `system_image` key in the JSON. Is there a list somewhere of acceptable inputs? What do the sections in that value mean? Most are self-explanatory, but not sure what `default` is referencing in the example in the docs.

Curious if there is any list for acceptable values for `device_type` and `skin` as well.

**Comment by rmartin16** [comment, marked as answer]
The simplest way to create arbitrary emulators is with Android Studio; it has a dedicated workflow with all the knobs and switches. You can specify the definition of the emulator when running Briefcase; there's an [example](https://github.com/beeware/briefcase/blob/9843f1ae4e96d06923e03a7b0ec2cd538189c86a/src/briefcase/integrations/android_sdk.py#L898) in the source comments.

Finally, Briefcase does not provide an interface to interact with AVDs it creates. However, they are in `~/.android`; I just delete them manually there.

**Reply by rmartin16** [reply]
Also, this is one way to produce a list of the available system images:
```
~/.cache/briefcase/tools/android_sdk/cmdline-tools/12.0/bin/sdkmanager --list
```
The location of [Briefcase's data directory](https://briefcase.readthedocs.io/en/stable/reference/environment.html#briefcase-home) is different on other platforms.

**Reply by rmartin16** [reply]
More generally, these are all just values to drive [`avdmanager`](https://developer.android.com/tools/avdmanager):
```
❯ ~/.cache/briefcase/tools/android_sdk/cmdline-tools/12.0/bin/avdmanager create
Error: Missing object name for verb 'create'.

Usage:
      avdmanager [global options] create [action options]
      Global options:
  -s --silent     : Silent mode, shows errors only.
  -v --verbose    : Verbose mode, shows errors, warnings and all messages.
     --clear-cache: Clear the SDK Manager repository manifest cache.
  -h --help       : Help on a specific command.

Valid actions are composed of a verb and an optional direct object:
- create avd          : Creates a new Android Virtual Device.

Action "create avd":
  Creates a new Android Virtual Device.
Options:
  -c --sdcard  : Path to a shared SD card image, or size of a new sdcard for
                 the new AVD.
  -g --tag     : The sys-img tag to use for the AVD. The default is to
                 auto-select if the platform has only one tag for its system
                 images.
  -p --path    : Directory where the new AVD will be created.
  -k --package : Package path of the system image for this AVD (e.g.
                 'system-images;android-19;google_apis;x86').
  -n --name    : Name of the new AVD. [required]
     --skin    : The optional name of a skin to use with this device.
  -f --force   : Forces creation (overwrites an existing AVD)
  -b --abi     : The ABI to use for the AVD. The default is to auto-select the
                 ABI if the platform has only one ABI for its system images.
  -d --device  : The optional device definition to use. Can be a device index
                 or id.
```

**Reply by ghost** [reply]
THANK YOU. The command to list system images is super clutch, needs to be in the docs I think.

I appreciate the extra info, things are making more sense now.

**Reply by freakboy3742** [reply]
The [AVD creation syntax *is* documented ](https://briefcase.readthedocs.io/en/stable/reference/platforms/android/gradle.html#d-device-device-device).

The longer term issue is #737 - allowing the "interactive" simulator creation process to allow the user to specify the AVD image, skin, etc.
